khal/controllers.py

```diff
@@ -585,3 +586,25 @@ def import_event(vevent, collection, locale, batch, format=None, env=None):
                 collection.force_update(Item(vevent), collection=calendar_name)
             else:
                 logger.warning("Not importing event with UID `{}`".format(event.uid))
+
+
+def print_ics(conf, name, ics, format):
+    if format is None:
+        format = conf['view']['agenda_event_format']
+    cal = icalendar.Calendar.from_ical(ics)
+    events = [item for item in cal.walk() if item.name == 'VEVENT']
+    events_grouped = defaultdict(list)
+    for event in events:
+        events_grouped[event['UID']].append(event)
+
+    if format is None:
+        format = conf['view']['event_format']
```

Comment:
I believe this second `if format is None` is superfluous, see line 591. Also according to the text in khal.spec `event_format` is the right format string, the current default just sucks.